rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
+    // The pointer is not at the beginning of a character - find the next one
+    while ((*str & 0xc0) == 0x80) { str++; }
```

Comment:
Will this run past the buffer in case a specially crafted string is made out of bytes that always pass the check?